packer/channels.scm

```diff
@@ -1,8 +1,8 @@
 (list (channel
     (name 'guix)
     (url "git://git.savannah.gnu.org/guix.git")
-    (branch "version-1.2.0")
+    ;; (tag "v1.3.0")
```

Comment:
Why the removal of this entirely? I'd get the setting of the `tag` instead of `branch` (given this is a static build of a particular release), but not commenting it out entirely?

Reply:
Good question!

I think I misunderstand channel usage with regards to git branch/tags, and this change (and it being weird for you) seems to be an offshoot of that.

If I understand correctly, channels describe source of packages and can pin the version of the package source, which means that `guix pull` will freeze packages to the versions that existed as of that commit/tag, not knowing of more recent ones.

This is good for reproducibility (hence probably usage in this case?) but annoying as a user, where one wants to "try pulling for latest and greatest". 

Personally, I'm still in the exploration phase of Guix, and haven't needed pinning versions much, I am more interested in seeing how up to date packages get (`racket@8.1`  was released on guix just a few days after the upstream release, great!). This meant I wanted to just point to `master` of guix, hence the removal (commenting out) of git tag.


So, do you mean for the branch/commit/tag to be for reproducibility? If so, I'll likely restore it. Otherwise, I am curious why have any branch specification (other than the `master` default)

Reply:
Yup, reproducibility. I'm trying to make a series of Virtualbox images that create a particular named version of GuixSD. If you OTOH want a "just follow master" image, then that's a different matter (and would probably require periodic rebuilds of the image which I'm not planning on doing given the manual overheads right now). However, what you can do is start from the most recent image I've provided and then replace the `channels.scm` in the running image. Or if you want to, make your own images, upload them to Vagrant Cloud and do whatever you like with those :)

Reply:
Hmm, yeah, that's what I thought. I was extrapolating from personal experience of "I want latest so have to modify the channels for master" into CI image building, which has totally different requirements. My bad.

I'll revert to the tag, then. No need to upload my latest-version image, it's just for fudging about locally really.